src/cargo/ops/cargo_install.rs

```diff
@@ -190,18 +201,23 @@ fn one<I, F>(mut i: I, f: F) -> CargoResult<Option<I::Item>>
 fn check_overwrites(dst: &Path,
                     pkg: &Package,
                     filter: &ops::CompileFilter,
-                    prev: &CrateListingV1) -> CargoResult<()> {
+                    prev: &CrateListingV1,
+                    replace: bool) -> CargoResult<HashMap<PackageId, Vec<String>>> {
     let check = |name| {
         let name = format!("{}{}", name, env::consts::EXE_SUFFIX);
         if fs::metadata(dst.join(&name)).is_err() {
-            return Ok(())
+            return Ok(None)
         }
         let mut msg = format!("binary `{}` already exists in destination", name);
         if let Some((p, _)) = prev.v1.iter().find(|&(_, v)| v.contains(&name)) {
-            msg.push_str(&format!(" as part of `{}`", p));
+            if replace {
+                return Ok(Some(p.clone()))
+            }
+            msg.push_str(&format!(" as part of `{}` (use --replace to override)", p));
         }
         Err(human(msg))
```

Comment:
I've realized that it's safer to always list all conflicts when run without `--replace` thus the error should probably be generated somewhere up the stack.